addons/barfiller/statics.lua

```diff
@@ -0,0 +1,433 @@
+--[[    BSD License Disclaimer
+        Copyright © 2015, Morath86
+        All rights reserved.
+
+        Redistribution and use in source and binary forms, with or without
+        modification, are permitted provided that the following conditions are met:
+
+            * Redistributions of source code must retain the above copyright
+              notice, this list of conditions and the following disclaimer.
+            * Redistributions in binary form must reproduce the above copyright
+              notice, this list of conditions and the following disclaimer in the
+              documentation and/or other materials provided with the distribution.
+            * Neither the name of BarFiller nor the
+              names of its contributors may be used to endorse or promote products
+              derived from this software without specific prior written permission.
+
+        THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+        ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+        WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+        DISCLAIMED. IN NO EVENT SHALL Morath86 BE LIABLE FOR ANY
+        DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+        (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+        LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+        ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+        (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+        SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+-- Default settings
+defaults = {
+    Bars = {
+        Background = {
+            Image = windower.windower_path..'addons\\barfiller\\data\\bar_bg.png',
```

Comment:
I second this. I highly suggest this be changed before the addon is merged. Windower *will not* overwrite or delete files in an addon's data directory. The directory doesn't belong to the addon, it belongs to the users. This was an intentional design decision so that addons could provide a safe location for user files that should not be overwritten during updates.

If you place images here you can't change them in the future. Things that an addon needs to function should not be in the data directory. Users should be able to delete the data directory and the addon should still be fully functional.

For the initial release I suggest using hard-coded paths (don't include them in settings) and moving the images into a separate directory, outside of the data directory. We can discuss the options you have for allowing customization for the next release with you on IRC if you want.